src/map/chat.hpp

```diff
@@ -5,42 +5,45 @@
 #define CHAT_HPP
 
 #include "map.hpp" // struct block_list, CHATROOM_TITLE_SIZE
+#include <array>
 
-class map_session_data;
-struct chat_data;
+inline constexpr int MAX_CHAT_USERS = 20;
+namespace chats{
 
-#define MAX_CHAT_USERS 20
-
-struct chat_data {
-	struct block_list bl;            // data for this map object
+class ChatData {
+public:
+	block_list bl;            // data for this map object
 	char title[CHATROOM_TITLE_SIZE]; // room title 
 	char pass[CHATROOM_PASS_SIZE];   // password
 	bool pub;                        // private/public flag
-	uint8 users;                     // current user count
-	uint8 limit;                     // join limit
-	uint8 trigger;                   // number of users needed to trigger event
+	uint32 trigger;                   // number of users needed to trigger event
+	uint32 users;                     // current user count
+	uint32 limit;                     // join limit
```

Comment:
Is there a reason why to increase the storage of these?
if it's to keep in line with MAX_CHAT_USERS, we can either change the storage type of MAX_CHAT_USERS, or have some static_assert comparing with `std::numeric_limit<decltype(ChatData::users)>::max()`

Reply:
client use uint16 here